integreat_cms/cms/utils/content_utils.py

```diff
@@ -0,0 +1,147 @@
+import logging
+from urllib.parse import urlparse
+
+from django.conf import settings
+from django.db.models import Q
+from lxml.html import HtmlElement, tostring
+
+from ..models import MediaFile
+from ..utils import internal_link_utils
+
+
+def clean_content(content: HtmlElement, language_slug: str) -> None:
+    """
+    This is the super function to clean content
+
+    :param content: the body of content that should be cleaned
+    :param language_slug: Slug of the current language
+    """
+    convert_heading(content)
+    convert_monospaced_tags(content)
+    update_links(content, language_slug)
+    process_images(content)
+
+
+def convert_heading(content: HtmlElement) -> None:
+    """
+    Converts every ``h1`` tag in the content to a ``h2`` for SEO purposes.
+
+    :param content: the body of content of which every ``h1`` should be converted to an ``h2``.
+    """
+    for heading in content.iter("h1"):
+        heading.tag = "h2"
+        logging.debug(
+            "Replaced heading 1 with heading 2: %r",
+            tostring(heading, encoding="unicode"),
+        )
+
+
+def convert_monospaced_tags(content: HtmlElement) -> None:
+    """
+    Converts ``pre`` and ``code`` tags to ``p`` tags.
+
+    :param content: the body of content of which every ``pre`` and ``code`` tag should be transformed
+    """
+    for monospaced in content.iter("pre", "code"):
+        tag_type = monospaced.tag
+        monospaced.tag = "p"
+        logging.debug(
+            "Replaced %r tag with p tag: %r",
+            tag_type,
+            tostring(monospaced, encoding="unicode"),
+        )
+
+
+def update_links(content: HtmlElement, language_slug: str) -> None:
+    """
+    Super method that gathers all methods related to updating links
+
+    :param content: The content whose links should be updated
+    :param language_slug: Slug of the current language
+    """
+    set_classes_for_links(content)
```

Comment:
Not too important I guess, but I think it would be nice if this function already contained the for loop and called the other functions from within that loop. That would save some processing work and would also be a bit more easy to understand I think.

Then this function could look roughly like this:
```python
    for link in content.iter("a"):
        set_classes_for_links(link)
        remove_target_attribute(link)
        update_internal_links(link, language_slug)
```